src/server/code-generator.ts

```diff
@@ -9,35 +9,55 @@ export class CodeGenerator extends EventTarget {
 
   private readonly destination: string;
 
+  private readonly generateOptions: { routes: boolean; types: boolean };
+
   private watcher: FSWatcher | undefined;
 
-  public constructor(openApiPath: string, destination: string) {
+  public constructor(
+    openApiPath: string,
+    destination: string,
+    generateOptions: { routes: boolean; types: boolean },
+  ) {
     super();
     this.openapiPath = openApiPath;
     this.destination = destination;
+    this.generateOptions = generateOptions;
   }
 
-  public async watch() {
-    await generate(this.openapiPath, this.destination);
+  public async generate() {
+    await generate(
+      this.openapiPath,
+      this.destination,
+      this.generateOptions.routes,
+      this.generateOptions.types,
+    );
+  }
 
+  public async watch() {
     if (this.openapiPath.startsWith("http")) {
```

Comment:
Why do we stop watching if the openapi doc is remote? What about watching the changes in the routes? Shouldn't this only return if we're only watching types?